cipher/cipher.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package cipher
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"fmt"
+	"sync"
+)
+
+type Mode string
+type Padding string
+
+const (
+	ModeEcb = Mode("ECB")
+)
+
+const (
+	PaddingPKCS7 = Padding("PKCS7Padding")
```

Comment:
we're only using PKCS5Padding, why add this?

Reply:
As I mentioned, PKCS#5 has 8-byte block, and PKCS #5 is defined for 8-byte block, while #7 is defined for various block size.
https://www.ietf.org/rfc/rfc2898.txt
In AES, in theory this should be PKCS#7, not #5. I believe your library implements #5 as #7 in AES case.